unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/WebSocketCommunication/WebSocketCommunication.cs

```diff
@@ -26,37 +26,66 @@ public class WebSocketCommunication : IKernelCommunication
 
     private string StartServer(int port, int maxPort)
     {
+        var debugConfig = GameObject.Find("DebugConfig").GetComponent<DebugConfigComponent>();
+
         if (port > maxPort)
         {
             throw new SocketException((int)SocketError.AddressAlreadyInUse);
         }
-        string wssServerUrl = $"ws://localhost:{port}/";
+        string wssServerUrl;
         string wssServiceId = "dcl";
-        string wssUrl = wssServerUrl + wssServiceId;
         try
         {
-            ws = new WebSocketServer(wssServerUrl);
+            if (debugConfig.webSocketSSL)
+            {
+                wssServerUrl = $"wss://localhost:{port}/";
+                ws = new WebSocketServer(wssServerUrl)
+                {
+                    SslConfiguration =
+                    {
+                        ServerCertificate = CertificateUtils.CreateSelfSignedCert(),
+                        ClientCertificateRequired = false,
+                        CheckCertificateRevocation = false,
+                        ClientCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true
+                    }
+                };                
+            }
+            else
+            {
+                wssServerUrl = $"ws://localhost:{port}/";
+                ws = new WebSocketServer(wssServerUrl);
+            }
+
             ws.AddWebSocketService<DCLWebSocketService>("/" + wssServiceId);
             ws.Start();
         }
         catch (InvalidOperationException e)
         {
             ws.Stop();
-            SocketException se = (SocketException)e.InnerException;
-            if (se is { SocketErrorCode: SocketError.AddressAlreadyInUse })
+            if (!debugConfig.webSocketSSL) // Don't search other ports if we're not using SSL
             {
-                return StartServer(port + 1, maxPort);
+                SocketException se = (SocketException)e.InnerException;
+                if (se is { SocketErrorCode: SocketError.AddressAlreadyInUse })
+                {
+                    return StartServer(port + 1, maxPort);
+                }
             }
             throw new InvalidOperationException(e.Message, e.InnerException);
         }
 
+        string wssUrl = wssServerUrl + wssServiceId;
         return wssUrl;
     }
 
     public WebSocketCommunication()
     {
         InitMessageTypeToBridgeName();
 
+        var debugConfig = GameObject.Find("DebugConfig").GetComponent<DebugConfigComponent>();
```

Comment:
i would try to find another way of getting `DebugConfigComponent` because finding objects by its name could lead into errors. You could have any other object called the same way in the scene or someone could change the name of it easily. It would be better to inject the component into `WebSocketCommunication` constructor. `DebugConfigComponent` could have its own singleton or a shared instance somewhere.
Also why you change the configuration here? I would be better to change it from `DebugConfigComponent` internally.